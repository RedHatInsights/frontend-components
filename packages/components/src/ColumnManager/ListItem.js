import React from 'react';
import propTypes from 'prop-types';
import { DataListCell, DataListCheck, DataListControl, DataListItem, DataListItemCells, DataListItemRow } from '@patternfly/react-core';

const ListItem = ({ column: { title }, onChange, isSelected, isDisabled, id, dataListCheckProps, dataListCellProps, children, ...props }) => (
  <DataListItem {...props}>
    <DataListItemRow>
      <DataListControl>
        <DataListCheck
          checked={isSelected}
          onChange={onChange}
          otherControls
          isDisabled={isDisabled}
          id={`${id}-id`}
          name={`${id}-id`}
          {...dataListCheckProps}
        />
      </DataListControl>
      <DataListItemCells
        dataListCells={[
          <DataListCell key={`${id}-cell`} {...dataListCellProps}>
            <label htmlFor={`${id}-id`}>{title}</label>
          </DataListCell>,
        ]}
      />
    </DataListItemRow>
    {children}
  </DataListItem>
);

ListItem.propTypes = {
  id: propTypes.string,
  column: propTypes.object,
  onChange: propTypes.func,
  isSelected: propTypes.bool,
  isDisabled: propTypes.bool,
  dataListCellProps: propTypes.object,
  dataListCheckProps: propTypes.object,
  children: propTypes.node,
};

export default ListItem;
