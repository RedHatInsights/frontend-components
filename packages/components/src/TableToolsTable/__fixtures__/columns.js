import { severityLevels } from './items';

export default [
  {
    title: 'ID',
    sortByProperty: 'id',
    exportKey: 'id',
  },
  {
    title: 'Name',
    sortByProperty: 'name',
    exportKey: 'name',
    renderFunc: (_a, _b, item) => item.name,
  },
  {
    title: 'Another Name column',
    sortByProperty: 'name',
    exportKey: 'name',
    renderExport: (name) => `${name} via export render`,
  },
  {
    title: 'Description',
    sortByProperty: 'description',
    renderExport: (item) => `${item.name} description rendered for export`,
  },
  {
    title: 'Severity sorted by function',
    sortByFunction: (item) => item.name,
    exportKey: 'severity',
    renderFunc: (_a, _b, item) => item.severity,
  },
  {
    title: 'Severity by Array',
    sortByProperty: 'severity',
    sortByArray: severityLevels,
    exportKey: 'severity',
    renderFunc: (_a, _b, item) => item.severity,
  },
];
