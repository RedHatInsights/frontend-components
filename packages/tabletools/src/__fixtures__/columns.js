import { severityLevels } from './items';

export default [
  {
    title: 'ID',
    sortByProperty: 'id',
    exportKey: 'id',
    key: 'id',
    managable: false,
  },
  {
    title: 'Name',
    key: 'id',
    sortByProperty: 'name',
    exportKey: 'name',
    renderFunc: (_a, _b, item) => item.name,
  },
  {
    title: 'Another Name column',
    key: 'another_name_column',
    sortByProperty: 'name',
    exportKey: 'name',
    renderExport: (name) => `${name} via export render`,
  },
  {
    title: 'Description',
    key: 'desc',
    sortByProperty: 'description',
    renderExport: (item) => `${item.name} description rendered for export`,
  },
  {
    title: 'Severity sorted by function',
    sortByFunction: (item) => item.name,
    exportKey: 'severity',
    renderFunc: (_a, _b, item) => item.severity,
  },
  {
    title: 'Severity by Array',
    sortByProperty: 'severity',
    sortByArray: severityLevels,
    exportKey: 'severity',
    renderFunc: (_a, _b, item) => item.severity,
  },
];
