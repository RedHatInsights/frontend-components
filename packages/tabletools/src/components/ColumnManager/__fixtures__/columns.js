export default [
  {
    title: 'ID',
    key: 'id',
  },
  {
    title: 'Another col',
    key: 'name',
    isRequired: true,
  },
  {
    title: 'Name',
    key: 'name1',
  },
  {
    title: 'Another Name column without key',
  },
];
