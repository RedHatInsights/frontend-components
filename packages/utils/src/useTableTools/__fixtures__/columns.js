import { severityLevels } from './items';

export default [
    {
        title: 'Name',
        key: 'name',
        sortByProperty: 'name',
        renderFunc: (item) => (
            item.name
        )
    }, {
        title: 'Description',
        key: 'description'
    }, {
        title: 'Severity sorted by function',
        sortByFunction: (item) => (severityLevels.indexOf(item.severity)),
        renderFunc: (item) => (
            item.severity
        )
    }, {
        title: 'Severity by Array',
        sortByProperty: 'severity',
        sortByArray: severityLevels,
        renderFunc: (item) => (
            item.severity
        )
    }
];
