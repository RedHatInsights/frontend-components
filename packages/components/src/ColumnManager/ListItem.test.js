import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import columns from './__fixtures__/columns';
import ListItem from './ListItem';

describe('ListItem', () => {
  it('expect to render without error', () => {
    expect(toJson(shallow(<ListItem column={columns[0]} />))).toMatchSnapshot();
  });
});
