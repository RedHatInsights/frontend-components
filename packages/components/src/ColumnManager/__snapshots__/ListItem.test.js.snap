// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem expect to render without error 1`] = `
<DataListItem
  aria-labelledby=""
  className=""
  id=""
  isExpanded={false}
>
  <DataListItemRow>
    <DataListControl>
      <DataListCheck
        id="undefined-id"
        name="undefined-id"
        otherControls={true}
      />
    </DataListControl>
    <DataListItemCells
      dataListCells={
        Array [
          <DataListCell>
            <label
              htmlFor="undefined-id"
            >
              ID
            </label>
          </DataListCell>,
        ]
      }
    />
  </DataListItemRow>
</DataListItem>
`;
